jerry-core/include/jerryscript-core.h

```diff
@@ -195,15 +195,16 @@ typedef jerry_value_t (*jerry_vm_exec_stop_callback_t) (void *user_p);
 typedef bool (*jerry_object_property_foreach_t) (const jerry_value_t property_name,
                                                  const jerry_value_t property_value,
                                                  void *user_data_p);
-/**
- * Function type for user context allocation
- */
-typedef void *(*jerry_user_context_init_t) (void);
 
 /**
- * Function type for user context deallocation
+ * User context item manager
  */
-typedef void (*jerry_user_context_deinit_t) (void *user_context_p);
+typedef struct
+{
+  void (*init_cb) (void *); /**< callback responsible for initializing a context item */
```

Comment:
Ok. However I think in most of the time people will simply zero the buffer, and this could be the default behaviour: pass a zeroed buffer to the init callback. If no init callback is provided (NULL), just zero the buffer.